src/xml.c

```diff
@@ -148,7 +148,7 @@ xhtml_extract_doc (xmlNodePtr xml, gint xhtmlMode, const gchar *defaultBase)
 				}
 				xmlFreeDoc (oldDoc);
 			}
-			xmlFree (escapedhtml);
+			g_free (escapedhtml);
```

Comment:
Why variable escapedhtml is currently freed using the g_free() instead of xmlFree()?
It is allocated by [xmlNodeListGetString()](http://xmlsoft.org/html/libxml-tree.html#xmlNodeListGetString) and it should be deallocated with xmlFree()

Reply:
There are 2 code paths one being xmlNodeListGetString() and one being g_strstrip(). This causes a type conflict, which is why the first code path casts to gchar. Therefore declaration and free'ing is of type gchar.

As for the background: you can configure libxml2 which allocator to use. And in Liferea it is configured to use Glib which is why gchar === xmlChar.

The reason why I changed the code above from xmlFree to g_free is to get rid of compiler warnings. Goal is to have a warning free compilation.

Reply:
> As for the background: you can configure libxml2 which allocator to use. And in Liferea it is configured to use Glib which is why gchar === xmlChar.

Ok, but if change an allocator to another one you have to remember to change g_free to a new function. If you forget this can lead to problems. 

> The reason why I changed the code above from xmlFree to g_free is to get rid of compiler warnings. Goal is to have a warning free compilation.

Is a type casting not enough? if I use xmlFree instead of g_free and xmlCharStrdup istead of g_strdup I don't see any warnings related to type conflicts.